src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingServiceTest.java

```diff
@@ -700,4 +800,296 @@ private ApiErrorResponseException getApiErrorResponseException() {
                 new HttpResponseException.Builder(401, "unauthorised", new HttpHeaders())
                         .setMessage(ERROR_MESSAGE));
     }
+
+    @Test
+    void testFilingGenerationWhenSuccessfulWithoutTrustsAndWithIdentityChecksWithTrustFeatureFlag() throws SubmissionNotFoundException, ServiceException, IOException, URIValidationException {
+        
+        setValidationEnabledFeatureFlag(true);
+
+        initTransactionPaymentLinkMocks();
+        initGetPaymentMocks();
+        when(localDateSupplier.get()).thenReturn(DUMMY_DATE);
+        ReflectionTestUtils.setField(filingsService, "filingDescriptionIdentifier", FILING_DESCRIPTION_IDENTIFIER);
+        ReflectionTestUtils.setField(filingsService, "filingDescription", FILING_DESCRIPTION);
+        OverseasEntitySubmissionDto overseasEntitySubmissionDto = Mocks.buildSubmissionDto();
+        Optional<OverseasEntitySubmissionDto> submissionOpt = Optional.of(overseasEntitySubmissionDto);
+        when(overseasEntitiesService.getOverseasEntitySubmission(OVERSEAS_ENTITY_ID)).thenReturn(submissionOpt);
+
+        FilingApi filing = filingsService.generateOverseasEntityFiling(OVERSEAS_ENTITY_ID, transaction, PASS_THROUGH_HEADER);
+
+        verify(localDateSupplier, times(1)).get();
+        assertEquals(FILING_KIND_OVERSEAS_ENTITY, filing.getKind());
```

Comment:
There's a certain amount of duplication from other tests here (e.g. checking field values that don't belong to trusts still match). Testing could in theory be narrowed down to just target the areas where the code changes were made in `FilingsService` but this is not a big issue.

Reply:
I will add a ticket to go back over our tests when we remove the feature flag from the code.

Reply:
I have asked for that ticket.